test/server/typegen.ts

```diff
@@ -367,7 +367,8 @@ test('typegen', async () => {
 
     export type Tables<
       PublicTableNameOrOptions extends
-        | keyof (Database["public"]["Tables"] & Database["public"]["Views"])
+        | keyof (Database[keyof Database]["Tables"] &
+            Database[keyof Database]["Views"])
```

Comment:
Hmm I think the concept here is to constrain each table to a new type on its own, for eg. `PublicTableNameOrOptions` type corresponds to the `public` schema exclusively.

We might need to generate separate `export type Tables` blocks depending on the schema name. @soedirgo wdyt?

Reply:
Ah sounds good! I was looking at that approach but since I don't personally use the new Table types I didn't know what the context of including them was. Happy to make this change (mapping over each schema, creating the PascalCase Table types, and updating the typegen types) and push up later this evening.

Reply:
@sweatybridge @soedirgo just pushed up some changes in order to create schema-specific types for Tables, TablesInsert, TablesUpdate, and Enums. Would love feedback on the source and test cases (should I add an additional test case to cover schemas not named public, and having multiple schemas)?

Reply:
After some thought, I think there's a backwards compatibility issue with this change. So I proposed a new approach in https://github.com/supabase/postgres-meta/pull/723 and added you as the co-author.